lib/node-repo.js

```diff
@@ -1,42 +1,88 @@
 'use strict'
 
+const request = require('request')
 const LRU = require('lru-cache')
 const retry = require('async').retry
+const debug = require('debug')('node_repo')
 
 const githubClient = require('./github-client')
 const resolveLabels = require('./node-labels').resolveLabels
+const { Owners } = require('./node-owners')
+const { createPrComment } = require('./github-comment')
 const existingLabelsCache = new LRU({ max: 1, maxAge: 1000 * 60 * 60 })
 
 const fiveSeconds = 5 * 1000
 
-function deferredResolveLabelsThenUpdatePr (options) {
-  const timeoutMillis = (options.timeoutInSec || 0) * 1000
-  setTimeout(resolveLabelsThenUpdatePr, timeoutMillis, options)
-}
-
-function resolveLabelsThenUpdatePr (options) {
+function getFilepathsChanged (options, handleFilepaths) {
   options.logger.debug('Fetching PR files for labelling')
 
-  const getFiles = (cb) => {
+  const getPRFiles = (cb) => {
     githubClient.pullRequests.getFiles({
       owner: options.owner,
       repo: options.repo,
       number: options.prId
     }, cb)
   }
 
-  retry({ times: 5, interval: fiveSeconds }, getFiles, (err, res) => {
+  retry({ times: 5, interval: fiveSeconds }, getPRFiles, (err, res) => {
     if (err) {
       return options.logger.error(err, 'Error retrieving files from GitHub')
     }
-
     const filepathsChanged = res.data.map((fileMeta) => fileMeta.filename)
-    const resolvedLabels = resolveLabels(filepathsChanged, options.baseBranch)
 
-    fetchExistingThenUpdatePr(options, resolvedLabels)
+    handleFilepaths(options, filepathsChanged)
   })
 }
 
+function deferredResolveOwnersThenPingPr (options) {
+  const timeoutMillis = (options.timeoutInSec || 0) * 1000
+  setTimeout(() => getFilepathsChanged(options, resolveOwnersThenPingPr), timeoutMillis, options)
+}
+
+function resolveOwnersThenPingPr (options, filepathsChanged) {
+  const { owner, repo } = options
+  githubClient.repos.get({
+    owner,
+    repo
+  }, (err, res) => {
+    if (err) {
+      return options.logger.error(err, 'Error retrieving repository data')
+    }
+    const data = res.data || {}
+    if (!data['default_branch']) {
+      return options.logger.error(err, 'Couldn\' determine default branch')
+    }
+
+    const { default_branch: defaultBranch } = data
+
+    const url = `https://raw.githubusercontent.com/${owner}/${repo}/${defaultBranch}/OWNERS.yml`
+    debug(`Fetching OWNERS on ${url}`)
+    request(url, (err, res, body) => {
+      if (err || !res || res.statusCode >= 400) {
+        return options.logger.error(err, 'Error retrieving OWNERS')
```

Comment:
Perhaps, also log `res` (or at least `res.statusCode`) in this case?

Reply:
Agreed, but can be done in a follow up PR (so we don't block the PR)